### ISSUE TITLE
Clicking links freeze undocked and unfocused frame5

### ISSUE THREAD

**Issue by BenCreating** [RPTools/maptool#2202]
**Describe the bug**
Clicking links in a floating frame5 causes the contents of the frame to freeze and stop responding if it did not have the focus.

**To Reproduce**
1. Create a library token called "Lib:Test"
2. Add a macro to the token, and call it "Test"
3. Paste this into the macro code:
`[frame5("Test Frame"): {[r: 1d100][r: macroLink("Roll Again", "Test@Lib:Test")]}]`
4. Run the macro to open the frame. Leave the frame undocked.
5. Click the link to confirm that it updates the number.
6. Click on the map to unfocus the frame.
7. Click the link. The number should stay the same, and the link should stop responding.

**Expected behavior**
The frame should gain focus, and execute the link without freezing.

**MapTool Info**
- Version: Tested in both 1.6.1 and 1.7.0
- Install: Upgrade from 1.5.10

**Desktop (please complete the following information):**
 - OS: MacOS
 - Version 10.15.6

**Additional context**
Frames can be unfrozen by clicking and dragging a link. Once you release it, the frame starts responding again.

**Comment by orlanthi** [comment]
Thank you so much for logging this issue @BenCreating I was starting to think I was doing something fundamentally
wrong with frames and macro links, especially as I couldn't see any reports on the forums of similar behaviour. The drag-the-link workaround means I can proceed as I was and hopefully there will be a fix relatively soon.

**Comment by Phergus** [comment]
Could not reproduce this under Windows with 1.7.0, 1.8.5 and current dev.  Need a Mac user to confirm this is still an issue.

**Comment by Azhrei** [comment]
Yes, it happens for me using both 183 and 185.  But it's complicated.

I have two monitors.  When I switch MT and the panel between the two monitors, the situation that fails is when both MT and the panel are on the second monitor.  If either one of them are on the first monitor, it works fine.

When the frame does lock up, I noticed that the link stays red (usually the link reverts to blue as soon as the event handler regains control).

I'm also getting an artifact window that appears at (0,0) on my main desktop if MT is on the second monitor when the frame opens.  (When MT is on the first monitor and I open the frame, it appears briefly and then goes away.)  It is the same size as the *Test Frame* but has no content and can't be interacted with (it has no border so can't be resized or moved).

I am on macOS 11.4 (fully updated) on a 2019 MBP.

**Comment by Phergus** [comment]
What about 1.9.0 or current dev?

**Comment by Azhrei** [comment]
Just tried 1.9.0 and had the exact same issues:  window placement on monitors had an effect.  I also went a little further this time and found that closing the panel and using the macro again opened the panel with a new random number, but the blue link couldn't be activated via mouse click.

While the panel is frozen, I edited the macro and changed the frame name, then ran it again.  It worked fine.  I do not see a spike in cpu activity while the panel is frozen, even with a half-dozen panels open.  (I did not try connecting via VisualVM or similar to look at thread activity.)

I could make some guesses about how the WebEngine is initialized for each frame and why that meant the second frame worked while the first was frozen, but that's best to someone looking at the code. :)

(Also noticed that dragging a macro from the GM panel to the Selection panel didn't provide the usual green "+" drop indicator, but that's a separate issue.)

**Comment by Phergus** [comment]
Then we'll be waiting still on a Mac developer to address this.
